designsafe/static/scripts/data-depot/components/published/published-view.component.js

```diff
@@ -35,6 +43,33 @@ class PublishedViewCtrl {
         this.browser.publication = this.publication;
         this.browser.project = this.publication.project;
         this.project = this.publication.project;
+        const { pi } = this.project.value;
+        this.UserService.get(pi).then((res) => {
+            this.authorData.pi = {
+                fname: res.first_name,
+                lname: res.last_name,
+                email: res.email,
+                name: res.username,
+                inst: res.profile.institution,
+            };
+            this.loadingUserData.pi = false;
+        });
+        if (this.project.value.coPis) {
+            this.authorData.coPis = new Array(this.project.value.coPis.length);
+            this.project.value.coPis.forEach((coPi, idx) => {
+                this.UserService.get(coPi).then((res) => {
```

Comment:
I believe there is a working method in the UserService for getting multiple user profiles at once called `getPublic` which should be able to accept a list of usernames.

Reply:
Oh lame! Yeah if the backend has that data and it's just not being passed up, see if you can add it in to the response. I'm sure it'll come in handy in the future. Sometimes we have like 20 or so users to look up on these projects and I don't really know how long it would take to grab each one.

If it turns out to be a bunch of extra work, just make a new JIRA task for it with the files and line numbers, then we'll come back to it.